istioctl/pkg/waypoint/waypoint.go

```diff
@@ -43,13 +50,95 @@ import (
 )
 
 var (
-	revision = ""
-
+	revision      = ""
+	waitReady     bool
+	waitTimeout   time.Duration
 	allNamespaces bool
+	revisionLabel = "istio.io/rev"
 
 	deleteAll bool
 )
 
+func isWaypointDeploymentReady(ctx cli.Context, kubeClient kube.CLIClient, gw *gateway.Gateway) (bool, error) {
+	deploymentName := fmt.Sprintf("%s-%s", gw.Name, gw.Spec.GatewayClassName)
+	deployment, err := kubeClient.Kube().AppsV1().Deployments(ctx.NamespaceOrDefault(ctx.Namespace())).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	// Check if all replicas are available and ready
+	if deployment.Status.AvailableReplicas == deployment.Status.Replicas {
+		return true, nil
+	}
+	return false, nil
+}
+
+type sidecarSyncStatus struct {
+	// nolint: structcheck, unused
+	pilot string
+	xds.SyncStatus
+}
+
+func xdsStatus(sent, acked string, typ model.NodeType) string {
+	if sent == "" {
+		if typ == model.Ztunnel {
+			return "IGNORED"
+		}
+		return "NOT SENT"
+	}
+	if sent == acked {
+		return "SYNCED"
+	}
+	// acked will be empty string when there is never Acknowledged
+	if acked == "" {
+		return "STALE"
+	}
+	// Since the Nonce changes to uuid, so there is no more any time diff info
+	return "STALE"
+}
+
+func isWaypointSynced(ctx cli.Context, kubeClient kube.CLIClient, gw *gateway.Gateway) (bool, error) {
```

Comment:
this is not needed. The pod cannot become ready without this being true

Reply:
Aaah, right. The deployment would not be ready if the initial configuration from the controller had not been received.
I removed the sync check logic as part of the new commits